linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -828,6 +828,13 @@ object Emitter {
               StringArgConstructorName)
         },
 
+        // See systemIdentityHashCode in CoreJSLib
+        callMethod(BoxedBooleanClass, hashCodeMethodName),
+        callMethod(BoxedDoubleClass, hashCodeMethodName),
+        callMethod(BoxedLongClass, hashCodeMethodName),
```

Comment:
Make this conditional on `useBigIntForLong`? (IIUC, if it is an object, we simply give it a normal identity hash code).